pkg/ebpf/c/common/context.h

```diff
@@ -209,6 +226,11 @@ statfunc int init_program_data(program_data_t *p, void *ctx)
     }
 
 out:
+    if (p->event->context.policies_version == 0) {
```

Comment:
Why do we have this condition?
Where did we ever initialized `p->event->context.policies_version` before?

Reply:
Since there are cases of calling `init_program_data()` (reentrant) and we already have set the policy_version and config via `init_policies_config()`. I've put this print to check such cases:

![image](https://github.com/aquasecurity/tracee/assets/3372117/f9167729-3050-4e10-be58-d81d13fe5b5a)

Ran: `sudo ./dist/tracee -e sched_process_exec`

And got this output:

```
...
          docker-616209  [011] ...2. 54180.265695: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 714
p->event->context.syscall 35

           <...>-616213  [031] ...2. 54180.265705: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 714
p->event->context.syscall 202

          docker-616215  [030] ...2. 54180.265709: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 0
p->event->context.syscall 202

           sleep-616207  [000] ...2. 54180.271460: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 714
p->event->context.syscall 231

     cpuUsage.sh-616201  [008] ...2. 54180.271677: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 0
p->event->context.syscall 56

     cpuUsage.sh-616216  [000] ...21 54180.271936: bpf_trace_printk: 
policies_version: 1
p->event->context.eventid 714
p->event->context.syscall 59
...
```

So I assume that it's working as a short-circuit.

Reply:
I think this is happening not due to it being reentrant, but because p->event is received from index 0 of a percpu map, which may have already been initialized from a previous event.

Actually this is even worse - after setting policies_version to 1, we will never be able to enter again the config init, thus we will never get a newer policy version

Reply:
You're right. I had concerns about tail calls calling again init_program_data, but it's not the case since they make use of `init_tailcall_program_data()` instead.